eclipse/src/saros/ui/wizards/AddReferencePointsToSessionWizard.java

```diff
@@ -610,6 +608,12 @@ private void showCancelMessage(JID jid, String errorMsg, CancelLocation cancelLo
       throw new IllegalStateException("no session running");
     }
 
+    IChecksumCache checksumCache = session.getComponent(IChecksumCache.class);
```

Comment:
Yes, that is true, but only because the `FileListFactory` checks whether the given checksum cache is `null` (which normally should never happen). It seems weird to me to rely on a nullcheck two methods deeper which is not even mentioned in the documentation (see [`createFileList`](https://github.com/saros-project/saros/blob/0cf2530c4d6b12a608e34153bc7f28f3577420b4/core/src/saros/negotiation/FileListFactory.java#L71) and [`calculateChecksums`](https://github.com/saros-project/saros/blob/0cf2530c4d6b12a608e34153bc7f28f3577420b4/core/src/saros/negotiation/FileListFactory.java#L145)).

Reply:
Yes it is not mentioned in the documentation.

As the whole checksum stuff is optional it is more a design choice if you force the usage of dummies or just allow `null` values.

Reply:
The checksum cache is not optional. As it is explicitly required by components in the session context (e.g. the resource negotiations), trying to create the session context without it leads to an UnsatisfiableDependenciesException: 
```
saros.negotiation.ResourceNegotiationFactory has unsatisfied dependency: interface saros.filesystem.checksum.IChecksumCache among unsatisfiable dependencies: [[class saros.net.xmpp.filetransfer.XMPPFileTransferManager, interface saros.editor.IEditorManager, class saros.observables.FileReplacementInProgressObservable, interface saros.filesystem.IWorkspace, interface saros.filesystem.checksum.IChecksumCache, interface saros.net.ITransmitter, interface saros.net.IReceiver, class saros.negotiation.AdditionalResourceDataFactory]] where saros.repackaged.picocontainer.DefaultPicoContainer@69283959:33<I<saros.repackaged.picocontainer.DefaultPicoContainer@224a4a5b:47<(empty) was the leaf container being asked for dependencies.
```

Reply:
You can avoid that by offering multiple CTORS.